x-pack/plugins/security_solution/public/timelines/components/flyout/index.tsx

```diff
@@ -30,13 +30,54 @@ interface OwnProps {
   onAppLeave: (handler: AppLeaveHandler) => void;
 }
 
+type VoidFunc = () => void;
+
 const FlyoutComponent: React.FC<OwnProps> = ({ timelineId, onAppLeave }) => {
   const dispatch = useDispatch();
   const getTimelineShowStatus = useMemo(() => getTimelineShowStatusByIdSelector(), []);
   const { activeTab, show, status: timelineStatus, updated } = useDeepEqualSelector((state) =>
     getTimelineShowStatus(state, timelineId)
   );
 
+  const [focusOwnership, setFocusOwnership] = useState(true);
+  const [triggerOnBlur, setTriggerOnBlur] = useState(true);
+  const callbackRef = useRef<VoidFunc | null>(null);
+  const searchRef = useRef<HTMLElement | null>(null);
+
+  const handleSearch = useCallback(() => {
+    if (show && focusOwnership === false) {
+      setFocusOwnership(true);
+    }
+  }, [show, focusOwnership]);
+  const onOutsideClick = useCallback((event) => {
+    setFocusOwnership(false);
+    const classes = event.target.classList;
+    if (classes.contains('euiFieldSearch')) {
+      searchRef.current = event.target;
+      setTriggerOnBlur((prev) => !prev);
```

Comment:
is this more `setTriggerOnStateChange`? All it's here to do is trigger the `useEffect` below right?